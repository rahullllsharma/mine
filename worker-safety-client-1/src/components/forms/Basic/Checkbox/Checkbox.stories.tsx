import type { ComponentMeta, ComponentStory } from "@storybook/react";

import { Checkbox } from "./Checkbox";

export default {
  title: "JSB/Checkbox",
  component: Checkbox,
} as ComponentMeta<typeof Checkbox>;

export const Playground: ComponentStory<typeof Checkbox> = args => (
  <Checkbox {...args}>Test</Checkbox>
);

Playground.args = {
  className: "mr-2",
};
