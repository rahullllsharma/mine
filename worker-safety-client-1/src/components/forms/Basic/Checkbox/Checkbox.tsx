import type { ReactNode } from "react";
import classnames from "classnames";

import { constVoid } from "fp-ts/lib/function";
import styles from "./Checkbox.module.css";

export type CheckboxProps = {
  className?: string;
  labelClassName?: string;
  // name: string;
  // value: string;
  // htmlFor: string;
  checked: boolean;
  disabled?: boolean;
  children?: ReactNode;
  // field: FormField<Errors, string, V>;
  onClick: () => void;
};

function Checkbox({
  className,
  labelClassName,
  // htmlFor,
  // name,
  // value,
  checked,
  disabled,
  // field,
  children,
  onClick,
}: CheckboxProps) {
  // const { raw, val, dirty } = field;
  // const hasError = dirty && isLeft(val);
  return (
    <>
      <div
        onClick={disabled ? constVoid : onClick}
        className={classnames("flex flex-row items-center gap-2", className)}
      >
        <input
          className={classnames(
            "flex-shrink-0",
            styles.root
            // { [styles.error]: hasError },
          )}
          // id={htmlFor}
          // name={name}
          // value={value}
          disabled={disabled}
          type="checkbox"
          checked={checked}
          readOnly
        />
        {/* <label htmlFor={htmlFor}>{children}</label> */}
        <label className={classnames("cursor-pointer", labelClassName)}>
          {children}
        </label>
      </div>

      {/* {hasError && <ErrorMessage field={field} />} */}
    </>
  );
}

export { Checkbox };
