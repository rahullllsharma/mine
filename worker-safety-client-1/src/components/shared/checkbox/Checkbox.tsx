import type { PropsWithClassName } from "@/types/Generic";

import type { ForwardedRef, HTMLProps } from "react";
import { forwardRef } from "react";
import cx from "classnames";

import styles from "./Checkbox.module.css";

export type CheckboxProps = HTMLProps<HTMLInputElement> &
  PropsWithClassName & {
    hasError?: boolean;
  };

function Checkbox(
  { className, hasError = false, ...props }: CheckboxProps,
  ref: ForwardedRef<HTMLInputElement>
) {
  return (
    <input
      {...props}
      type="checkbox"
      className={cx(
        "flex-shrink-0",
        { "!border-system-error-40": hasError },
        styles.root,
        className
      )}
      ref={ref}
    />
  );
}

export default forwardRef(Checkbox);
