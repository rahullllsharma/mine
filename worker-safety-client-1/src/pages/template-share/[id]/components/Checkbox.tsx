import type { CheckboxQuestionPropertiesType } from "@/components/templatesComponents/customisedForm.types";
import { BodyText, ComponentLabel } from "@urbint/silica";
import Link from "@/components/shared/link/Link";

const isEmptyValue = (value: any): boolean => {
  if (value === null || value === undefined) return true;
  if (Array.isArray(value) && value.length === 0) return true;
  return false;
};

const Checkbox = ({
  checkboxProperties,
}: {
  checkboxProperties: CheckboxQuestionPropertiesType;
}) => {
  const { url, url_display_text } = checkboxProperties?.options?.[0] || {};

  if (isEmptyValue(checkboxProperties?.user_value)) {
    return (
      <div className="flex flex-col gap-2 bg-brand-gray-10 p-4 rounded-lg">
        <ComponentLabel className="text-sm font-semibold">
          {checkboxProperties.title}
        </ComponentLabel>
        <BodyText className="text-base text-neutrals-tertiary">
          No information provided
        </BodyText>
      </div>
    );
  }

  return (
    <>
      {checkboxProperties?.user_value?.includes("1") ? (
        <div className="flex flex-col gap-2 bg-brand-gray-10 p-4 rounded-lg">
          <BodyText className="text-base">{checkboxProperties.title}</BodyText>
          {url && url_display_text && (
            <Link
              href={url}
              target="_blank"
              rel="noopener noreferrer"
              label={url_display_text}
              iconRight="external_link"
              className="text-base font-normal"
            />
          )}
        </div>
      ) : null}
    </>
  );
};

export default Checkbox;
