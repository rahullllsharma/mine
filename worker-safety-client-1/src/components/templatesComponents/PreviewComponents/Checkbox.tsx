import type { CheckboxPreviewProps } from "../customisedForm.types";
import React, { useState, useEffect } from "react";
import { BodyText, CaptionText } from "@urbint/silica";
import {
  FORM_EVENTS,
  formEventEmitter,
} from "@/context/CustomisedDataContext/CustomisedFormStateProvider";
import Checkbox from "@/components/shared/checkbox/Checkbox";
import Link from "@/components/shared/link/Link";
import { disabledField } from "../customisedForm.utils";
import SvgButton from "../ButtonComponents/SvgButton/SvgButton";

const CheckboxPreview: React.FC<CheckboxPreviewProps> = ({
  content,
  onChange,
  mode,
  inSummary,
}) => {
  const {
    title,
    is_mandatory,
    user_value = [],
    options = [],
    include_in_widget,
  } = content.properties;
  const url = options.length > 0 ? options[0].url : "";
  const url_display_text =
    options.length > 0 ? options[0].url_display_text : "";
  const [localValue, setLocalValue] = useState<string[]>(user_value || []);
  const [error, setError] = useState(false);

  const isChecked = localValue.includes("1");

  useEffect(() => {
    onChange(localValue);
  }, [localValue]);

  useEffect(() => {
    const token = formEventEmitter.addListener(
      FORM_EVENTS.SAVE_AND_CONTINUE,
      () => {
        if (is_mandatory && !isChecked) {
          setError(true);
        } else {
          setError(false);
          onChange(localValue);
        }
      }
    );
    return () => {
      token.remove();
    };
  }, [localValue, is_mandatory, isChecked, onChange]);

  useEffect(() => {
    if (error) {
      if (localValue.includes("1")) {
        setError(false);
      } else {
        setError(true);
      }
    }
  }, [error, localValue]);

  const handleCheckboxChange = () => {
    setLocalValue(prev => (prev.includes("1") ? [] : ["1"]));
  };

  return (
    <div className="flex flex-col gap-2" id={content.id}>
      <div className="flex items-center gap-2">
        {!inSummary && (
          <Checkbox
            checked={isChecked}
            onChange={handleCheckboxChange}
            disabled={disabledField(mode)}
            hasError={error}
          />
        )}

        <div className="flex gap-2">
          <BodyText className="text-base">
            {title}
            {is_mandatory && !inSummary && (
              <span className="text-black ml-1">*</span>
            )}
          </BodyText>
          {include_in_widget && mode === "BUILD" && (
            <div className="text-neutrals-tertiary flex gap-2">
              <SvgButton svgPath={"/assets/CWF/widget.svg"} />
              <BodyText className="text-neutrals-tertiary">Widget</BodyText>
            </div>
          )}
        </div>
      </div>
      {error && (
        <CaptionText className="text-red-500 text-sm mt-1 ml-7 cursor-auto">
          This is required
        </CaptionText>
      )}
      {url && url_display_text && (
        <Link
          href={url}
          target="_blank"
          rel="noopener noreferrer"
          label={url_display_text}
          iconRight="external_link"
          className={`text-base mt-1 ${
            inSummary ? "font-semibold" : "font-normal ml-7"
          }`}
          style={{
            wordBreak: "break-word",
          }}
        />
      )}
    </div>
  );
};

export default CheckboxPreview;
