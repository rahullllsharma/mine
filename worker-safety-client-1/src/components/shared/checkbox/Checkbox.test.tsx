import { fireEvent, render, screen } from "@testing-library/react";
import Checkbox from "./Checkbox";

describe(Checkbox.name, () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Checkbox />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should accept a default value", () => {
    render(<Checkbox defaultChecked />);
    screen.getByRole("checkbox", { checked: true });
  });

  it("should trigger an onChange event when clicked", () => {
    const mockOnChange = jest.fn();
    render(<Checkbox onChange={mockOnChange} checked />);
    screen.getByRole("checkbox", { checked: true });

    fireEvent.click(screen.getByRole("checkbox"));
    expect(mockOnChange).toHaveBeenCalled();
  });
});
