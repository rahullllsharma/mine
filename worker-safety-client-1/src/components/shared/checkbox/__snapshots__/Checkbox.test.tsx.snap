// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`undefined should render correctly 1`] = `
<DocumentFragment>
  <input
    class="flex-shrink-0 root"
    type="checkbox"
  />
</DocumentFragment>
`;
